### PR TITLE
[chore] Refactor `FeatureModel` & `FeatureNamespaceModel` for target models

### PR DESCRIPTION
**Pull request by gee-senbong**
## Description

As the title suggested, this PR refactor these feature models so that those common functionality could be shared with target models. This PR does not introduce any new functionality to existing code base.

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/models/feature.py [hunk 1]
```diff
@@ -7,213 +7,46 @@
...
     table_ids: List[PydanticObjectId] = Field(allow_mutation=False, default_factory=list)
     primary_table_ids: List[PydanticObjectId] = Field(allow_mutation=False, default_factory=list)
-    feature_namespace_id: PydanticObjectId = Field(allow_mutation=False, default_factory=ObjectId)
-    feature_list_ids: List[PydanticObjectId] = Field(allow_mutation=False, default_factory=list)
-    deployed_feature_list_ids: List[PydanticObjectId] = Field(
-        allow_mutation=False, default_factory=list
-    )
     user_defined_function_ids: List[PydanticObjectId] = Field(
         allow_mutation=False, default_factory=list
     )
```
Comment (gee-senbong) [inline featurebyte/models/feature.py:52]: @jevonyeoh @YS-L 
These are the common attributes that should share between target & feature. Does it make sense to you?

**Comment by gee-senbong** [comment]
Note that the changes in documentation is due to the changes of `FeatureReadiness` & `DefaultVersionMode` import paths.

File: featurebyte/__init__.py [hunk 2]
```diff
@@ -435,7 +435,6 @@ def list_deployments(
     "ValueBeyondEndpointImputation",
     "StringValueImputation",
     # feature & feature list version specific classes
-    "DefaultVersionMode",
```
Comment (jevonyeoh) [inline featurebyte/__init__.py:438]: curious what's the reason behind removing this?

File: featurebyte/models/feature.py (hunk 1 above)
Comment (jevonyeoh) [reply featurebyte/models/feature.py:52]: makes sense to me!

**Review by jevonyeoh — APPROVED** [review]

File: featurebyte/__init__.py (hunk 2 above)
Comment (gee-senbong) [reply featurebyte/__init__.py]: Just found this unexpected change. Has reverted the changes here.

**Pull request merged**